File: RB110/small_problems/medium2/unlucky_days.rb
=begin

Write a method that returns the number of Friday the 13ths in the year given by an argument. 
You may assume that the year is greater than 1752 (when the United Kingdom adopted the modern
 Gregorian Calendar) and that it will remain in use for the foreseeable future.


Input:
  -Integer representing a year

Ouput:
  -An integer representing how many fridays fall on the 13th of each month in the given year

Rules:
  -Explicit:
    -Assume the year is greater then 1752
  -Implicit
    -Return an integer

  -Questions?:
    -Will there be an input year greater than the year we are in?

Test cases / Exsamples:

friday_13th(2015) == 3
friday_13th(1986) == 1
friday_13th(2019) == 2

Data structure:
  -integers

Algorithm:
  -Set a day variable to 0
  -Set a friday the 13th variable to 0
  -Loop while day is less than 365
    -Check if day is friday an also the 13th?
    -if true increment the friday the 13th variable
  -Return friday the 13th variable

=end


require 'date'

def friday_13th(year)
  month = 1
  friday_13th = 0

  while month <= 12
    if Date.new(year, month, 13).friday?
      friday_13th += 1
    end
    month += 1
  end
  
  p friday_13th
end

friday_13th(2015) #== 3
friday_13th(1986) #== 1
friday_13th(2019) #== 2

File: LS215/exerises/medium2/unlucky_days.js
/*
Write a function that takes a year as an argument and returns the number of 'Friday the 13ths' in that year. 
You may assume that the year is greater than 1752 (when the modern Gregorian Calendar was adopted by the 
United Kingdom). You may also assume that the same calendar will remain in use for the foreseeable future.

Input:
  - An integer representing the year

Output:
  - An integer represening how many fridays the 13ths are in the input year

Rules:
  - The year will be greater then 1752
  - Use the Gregorian calendar
  - Return the number of friday the 13ths for that year
  
Questions?
  - Will there be an input other then an integers?
  - Will there be more then one argument input? No arguments?

Data Structure
  - Date object
  - Integers 
  - Strings

Algorithm:
  - Grab the date object of the input year with the month starting at 0 and day of 13
  - create a counter
  - Loop through 0 - 11 (months)
    - Us the index to increment the months
      - get the day of the week
      - If the day of the week id === 5 (friday)
        - increment the counter
  - return counter

*/

// let date = new Date(1986, 1, 13);

// console.log(date.getDay() === 5);

function fridayThe13ths(year) {
  let counter = 0;
  

  for (let i = 0; i < 12; i++) {
    let date = new Date(year, i, 13)

    if (date.getDay() === 5) {
      counter++
    }
  }
  return counter;
}

console.log(fridayThe13ths(1986));      // 1
console.log(fridayThe13ths(2015));      // 3
console.log(fridayThe13ths(2017));      // 2